lib/ssh_scan/scan_engine.rb

```diff
@@ -9,11 +9,34 @@ def scan_target(socket, opts)
       target, port = socket.chomp.split(':')
       policy = opts[:policy_file]
       timeout = opts[:timeout]
+      result = []
 
-      client = SSHScan::Client.new(target, port, timeout)
-      client.connect()
-      result = client.get_kex_result()
-      return result if result.include?(:error)
+      if target.fqdn?
+        if target.resolve_fqdn_as_ipv6.nil?
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv4.to_s, port, timeout)
+          client.connect()
+          result = client.get_kex_result()
+          result[:hostname] = target
+          return result if result.include?(:error)
+        else
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv6.to_s, port, timeout)
```

Comment:
@jinankjain from the perspective of the scanner, in order for it to talk to IPv4 and IPv6 targets, the scanner itself will need to support IPv4 and IPv6 networking.  It's pretty common these days to see most hosts support IPv4, but not IPv6.  However, our DNS queries for "github.com" for the A and AAAA DNS records will happen over whatever the default is for that host.  So if the host does not have IPv6 networking, but it can still do a DNS lookup for a AAAA record and get an IPv6 address what we expect to happen?